src/Ast.ml

```diff
@@ -217,8 +217,67 @@ let longident_is_simple c x =
 let module_type_is_simple x =
   match x.pmty_desc with Pmty_signature _ -> false | _ -> true
 
-let module_expr_is_simple x =
-  match x.pmod_desc with Pmod_structure _ -> false | _ -> true
+let rec module_expr_is_simple x =
+  match x.pmod_desc with
+  | Pmod_structure _ -> false
+  | Pmod_constraint (e, t) ->
+      module_expr_is_simple e && module_type_is_simple t
+  | _ -> true
```

Comment:
You should check `Pmod_functor` and `Pmod_apply`, they can contain sub-expressions (and types).